framework/src/executioners/Transient.C

```diff
@@ -741,6 +741,16 @@ Transient::getTimeStepperName()
     return std::string();
 }
 
+std::string
+Transient::getTimeIntegratorName() const
+{
+  const auto * ti = _nl.getTimeIntegrator();
+  if (ti)
+    return ti->type();
+  else
+    mooseError("Time integrator has not been built yet so we can't retrieve its name");
```

Comment:
```suggestion
  if (const auto * ti = _nl.getTimeIntegrator())
    return ti->type();
  return {};
```

I don't think it's necessary to error here - your call

Reply:
yeah either/or. with an error we re sure to never print the wrong thing